### PR TITLE
Add `namedtuple` to `BaseAnnotation` bases

### PR DESCRIPTION
**Pull request by madkinsz**
Extends https://github.com/PrefectHQ/prefect/pull/7263 allowing annotations to be serialized/deserialized by libraries that support namedtuples.

With https://github.com/dask/distributed/pull/7292 this will close https://github.com/PrefectHQ/prefect-dask/issues/43

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`

### CODE REVIEW

**Comment by ahuang11** [comment]
Unsure why Python 3.7 fails but all other versions are fine
```

ERROR    prefect.flow_runs:engine.py:399 Finished in state Failed('Flow run encountered an exception. RecursionError: maximum recursion depth exceeded in comparison\n')
_________________ TestUnmapped.test_always_returns_same_value __________________
[gw0] linux -- Python 3.7.15 /opt/hostedtoolcache/Python/3.7.15/x64/bin/python

self = <tests.utilities.test_annotations.TestUnmapped object at 0x7f2433bedc10>

    def test_always_returns_same_value(self):
        thing = unmapped("hello")
    
        for _ in range(10):
>           assert thing[random.randint(0, 100)] == "hello"

tests/utilities/test_annotations.py:11: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/prefect/utilities/annotations.py:44: in __getitem__
    return self.unwrap()
src/prefect/utilities/annotations.py:20: in unwrap
    return self.value
src/prefect/utilities/annotations.py:44: in __getitem__
    return self.unwrap()
src/prefect/utilities/annotations.py:20: in unwrap
    return self.value
E   RecursionError: maximum recursion depth exceeded in comparison
```

**Comment by ahuang11** [comment]
```
    def __getitem__(self, _) -> T:
        # Internally, this acts as an infinite array where all items are the same value
        return super().__getitem__(_)
```
seems to have fixed the recursion error and returns the expected result. Will see if this breaks other tests.

**Comment by ahuang11** [comment]
Okay I think I found a workaround by converting to dict first; not sure if this should only apply for python 3.7
```
    def unwrap(self) -> T:
        # cannot simply return self.value due to recursion error in Python 3.7
        # also _asdict does not follow convention; it's not an internal method
        # https://stackoverflow.com/a/26180604
        return self._asdict()["value"]
```

File: src/prefect/utilities/annotations.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ class unmapped(BaseAnnotation[T]):
 
     def __getitem__(self, _) -> T:
         # Internally, this acts as an infinite array where all items are the same value
-        return self.unwrap()
+        return super().__getitem__(_)
```
Comment (madkinsz) [inline src/prefect/utilities/annotations.py]: mm this isn't what we want to do  here `__getitem__` should return the value no matter the index. Is this being called by `self.value`?

Comment (ahuang11) [reply src/prefect/utilities/annotations.py]: Right, I think you might have to refresh to see the new changes.

**Comment by ahuang11** [comment]
Yes I've considered it since I'm not sure if using asdict is costly.

File: tests/utilities/test_collections.py [hunk 2]
```diff
@@ -408,6 +408,13 @@ def test_visit_collection_max_depth(self, inp, depth, expected):
         assert result == expected
 
     def test_visit_collection_context(self):
+        foo = [1, 2, [3, 4], [5, [6, 7]], 8, 9]
+
+        def visit(expr, context):
+            if isinstance(expr, list):
+                context["depth"] += 1
+                return expr
+
```
Comment (ahuang11) [inline tests/utilities/test_collections.py:417]:
```suggestion
```

**Comment by ahuang11** [comment]
Ok, I think this should be ready.

File: src/prefect/utilities/annotations.py [hunk 3]
```diff
@@ -1,20 +1,32 @@
...
+            # https://stackoverflow.com/a/26180604
+            return self._asdict()["value"]
+        else:
+            return self.value
+
+    def rewrap(self, value: T) -> "BaseAnnotation[T]":
+        return type(self)(value)
 
     def rewrap(self, value: T) -> "BaseAnnotation[T]":
         return type(self)(value)
```
Comment (madkinsz) [inline src/prefect/utilities/annotations.py:32]: @ahuang11 looks like this got duplicated

Comment (ahuang11) [reply src/prefect/utilities/annotations.py]: Thanks; fixed it.

**Pull request merged**